kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -352,9 +353,13 @@ func fromKafkaMessage(ctx context.Context, kafkaMessage *sarama.ConsumerMessage)
 		case "ce_dataschema":
 			event.SetDataSchema(v)
 		default:
-			// Extensions
-			if IsAlphaNumeric(h) {
-				event.SetExtension(h, v)
+			// Extensions. Strip ce_ first before calling IsAlphaNumeric
+			// because _ is not valid...
+			// TODO: Should we even allow headers to be added as extensions
```

Comment:
Not sure we should add normal kafka headers in this way, KafkaSource behaves differently: https://github.com/knative/eventing-contrib/blob/master/kafka/source/pkg/adapter/adapter.go#L280 . For now I would prefer to remove it, as it creates another inconsistency

Reply:
so, if the header does not begin with ce_ do NOT add it?

Reply:
yes OR, as i do in kafka source, prepend it as ce_kafkaheader

Reply:
@slinkydeveloper I have hopes, that w/ the Kafka support directly on the CloudEvent SDK, we get rid of all of this crap